### PR TITLE
Add notes on the project code philosophy

### LINKED ISSUE #833: Add short section to the overview chapter covering the sample code
In particular, we should stress that our code is a toy implementation, isn't a recommended C++ implementation, leaves a lot of features on the table, and so forth.

### PR DESCRIPTION
**Pull request by hollasch**
Resolves #833

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,9 +24,9 @@ The easiest way to help out is to log any issues you find in the books. Unclear
 all kinds, even better ways to present something -- just go to the [issues page][].
 
 **Before creating a new issue**, please review existing issues to see if someone has already
-submitted the same one. Odds are you're not the first to encounter something, so a little quick
+submitted the same one. Chances are you're not the first to encounter something, so a little quick
 research can save everyone some hassle. It's also a good idea to verify that problems still exist in
-the `development` branch when creating new issues.
+the `dev-xxxxx` branch when creating new issues.
```
Comment (trevordblack) [inline CONTRIBUTING.md:29]: "in the active development branch, which will be one of `dev-major`, `dev-minor`, or `dev-patch`, when creating new issues

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -55,6 +55,12 @@ To contribute a change to the project, *please follow these steps*:
 
   7. When ready, create your pull request and request a review from "rt-contributors".
 
+Note the code philosophy outlined in the first section of the first book. In short, the code is
+intended to be clear for everyone, using as little C/C++ arcania as possible. We have chosen to
```
Comment (trevordblack) [inline CONTRIBUTING.md:59]: [sp]? arcana?

Esoterica? C/C++isms ?

File: books/RayTracingInOneWeekend.html [hunk 3]
```diff
@@ -45,9 +45,28 @@
...
+
+The code thus provides a baseline implementation, with tons of improvements left for the reader to
+enjoy. There are endless ways one can optimize and modernize the code; we prioritize the simple
+solution.
+
+We assume a little bit of familiarity with vectors (like dot product and vector addition). If you
 don’t know that, do a little review. If you need that review, or to learn it for the first time,
-check out Marschner’s and my graphics text, Foley, Van Dam, _et al._, or McGuire’s graphics codex.
+check out Marschner’s and Peter's graphics text, Foley, Van Dam, _et al._, or McGuire’s graphics
+codex.
```
Comment (trevordblack) [inline books/RayTracingInOneWeekend.html:69]: Perhaps include ISBNs and a link to the codex?

Not necessary I think

**Review by trevordblack — APPROVED** [review]

**Review by trevordblack — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (hollasch) [reply CONTRIBUTING.md:59]: Changed to "using simple C/C++ idioms as much as possible."

File: CONTRIBUTING.md (hunk 1 above)
Comment (hollasch) [reply CONTRIBUTING.md:29]: Changed to "still exist in the development branch (which will be one of `dev-patch`, `dev-minor` or `dev-major`) when creating new issues."

File: books/RayTracingInOneWeekend.html (hunk 3 above)
Comment (hollasch) [reply books/RayTracingInOneWeekend.html:69]: Changed to:

> We assume a little bit of familiarity with vectors (like dot product and vector addition). If you
> don’t know that, do a little review. If you need that review, or to learn it for the first time,
> check out the online [_Graphics Codex_][gfx-codex] by Morgan McGuire, _Fundamentals of Computer
> Graphics_ by Steve Marschner and Peter Shirley, or _Fundamentals of Interactive Computer Graphics_
> by J.D. Foley and Andy Van Dam.

**Pull request merged**